### PR TITLE
Add Mechacomic

### PR DESCRIPTION
**Pull request by MikeZeDev**
https://github.com/manga-download/hakuneko/issues/7343

### CODE REVIEW

File: web/src/engine/websites/MechaComic.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                .join(' ').trim();
+            const link = chapter.querySelector<HTMLAnchorElement>('a');
+            return new Chapter(this, manga, link.pathname, title);
+        });
+    }
+
+    public override async FetchPages(chapter: Chapter): Promise<Page[]> {
+        const chapterUrl = new URL(chapter.Identifier, this.URI);
+        const response = await Fetch(new Request(chapterUrl));
+        const url = response.redirected ? new URL(response.url) : chapterUrl;
```
Comment (ronny1982) [inline web/src/engine/websites/MechaComic.ts:54]: This check seems redundant, isn't `response.url` always the correct URL that should be used?

File: web/src/engine/websites/MechaComic.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+        const rasterScriptURL = url.searchParams.get('contents');
+        const verticalScriptURL = url.searchParams.get('contents_vertical');
+        const pageScriptURL = url.searchParams.get('contents_page');
+        const cryptokeyURL = url.searchParams.get('cryptokey');
+        const baseUrl = url.searchParams.get('directory');
+        const ver = url.searchParams.get('ver');
+
+        //Fetch key
```
Comment (ronny1982) [inline web/src/engine/websites/MechaComic.ts:63]: Redundant comment can be removed

File: web/src/engine/websites/MechaComic.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        const cryptoKey = cryptokeyURL ? await (await Fetch(new Request(new URL(cryptokeyURL, this.URI)))).text() : '';
+
+        //Fetch content.json
```
Comment (ronny1982) [inline web/src/engine/websites/MechaComic.ts:66]: Redundant comment can be removed

File: web/src/engine/websites/MechaComic.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        const contentUrl = new URL(rasterScriptURL || verticalScriptURL || pageScriptURL, chapterUrl);
+        contentUrl.searchParams.set('ver', ver);
+        const { images } = await FetchJSON<ContentData>(new Request(contentUrl));
+        return Object.values(images).map(page => {
+            const url = new URL(page.shift().src, baseUrl);
+            url.searchParams.set('ver', ver);
+            return new Page(this, chapter, url, { cryptoKey });
```
Comment (ronny1982) [inline web/src/engine/websites/MechaComic.ts:73]: Consider use latest introduction of generic typing for page parameters

File: web/src/engine/websites/MechaComic.ts [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        });
+    }
+
+    public override async FetchImage(page: Page, priority: Priority, signal: AbortSignal): Promise<Blob> {
```
Comment (ronny1982) [inline web/src/engine/websites/MechaComic.ts:77]: Consider use latest introduction of generic typing for page parameters

**Review by ronny1982 — APPROVED** [review]

File: web/src/engine/websites/MechaComic.ts (hunk 1 above)
Comment (MikeZeDev) [reply web/src/engine/websites/MechaComic.ts:54]: Yes but if no redirection occurs (who knows with those websites) response.url will undefined.
This way is safer, and its a simple check.

**Pull request merged**